rust_src/src/base64.rs

```diff
@@ -0,0 +1,127 @@
+extern crate libc;
+extern crate base64;
+
+use base64::base64::{decode, encode};
+use std::ffi::CStr;
+use std::ffi::CString;
+use std::ptr;
+
+const MIME_LINE_LENGTH: usize = 76;
```

Comment:
I am wondering; maybe the whole definition should be in strings... I don't know. I feel maybe it shouldn't be in its own file.